tests/Unit/PointwiseFunctions/Xcts/Test_AdmLinearMomentum.cpp

```diff
@@ -3,87 +3,74 @@
-#include "Domain/CoordinateMaps/CoordinateMap.hpp"
-#include "Domain/CoordinateMaps/CoordinateMap.tpp"
-#include "Domain/CoordinateMaps/KerrHorizonConforming.hpp"
-#include "Domain/CoordinateMaps/Wedge.hpp"
 #include "Domain/CreateInitialElement.hpp"
 #include "Domain/Creators/Sphere.hpp"
 #include "Domain/ElementMap.hpp"
 #include "Domain/ElementToBlockLogicalMap.hpp"
 #include "Domain/FaceNormal.hpp"
 #include "Domain/InterfaceLogicalCoordinates.hpp"
-#include "Domain/Structure/IndexToSliceAt.hpp"
 #include "Domain/Structure/InitialElementIds.hpp"
 #include "Framework/CheckWithRandomValues.hpp"
 #include "Framework/SetupLocalPythonEnvironment.hpp"
 #include "NumericalAlgorithms/LinearOperators/DefiniteIntegral.hpp"
-#include "NumericalAlgorithms/LinearOperators/Divergence.hpp"
 #include "NumericalAlgorithms/Spectral/LogicalCoordinates.hpp"
 #include "NumericalAlgorithms/Spectral/Mesh.hpp"
-#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/HarmonicSchwarzschild.hpp"
-#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrHorizon.hpp"
 #include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp"
 #include "PointwiseFunctions/AnalyticSolutions/Xcts/Schwarzschild.hpp"
 #include "PointwiseFunctions/AnalyticSolutions/Xcts/WrappedGr.hpp"
 #include "PointwiseFunctions/Xcts/AdmLinearMomentum.hpp"
 
 namespace {
 
+using Schwarzschild = Xcts::Solutions::Schwarzschild;
 using KerrSchild = Xcts::Solutions::WrappedGr<gr::Solutions::KerrSchild>;
 
-void test_linear_momentum_surface_integral(const double distance,
-                                           const size_t refinement_level,
-                                           const size_t polynomial_order) {
-  // Define black hole parameters
-  const double mass = 1;
-  const double boost_speed = 0.5;
-  const double lorentz_factor = 1. / sqrt(1. - square(boost_speed));
-
-  // Get Kerr-Schild solution
-  const std::array<double, 3> dimensionless_spin{{0., 0., 0.}};
-  const std::array<double, 3> boost_velocity{{0., 0., boost_speed}};
-  const KerrSchild solution{mass, dimensionless_spin,
-                            /* center */ std::array<double, 3>{{0., 0., 0.}},
-                            boost_velocity};
+template <typename... RequestedTags>
+tuples::TaggedTuple<RequestedTags...> get_solution_variables(
+    const std::variant<Schwarzschild, KerrSchild>& solution,
+    const tnsr::I<DataVector, 3, Frame::Inertial>& coords,
+    const tmpl::list<RequestedTags...> tags) {
+  if (auto* schwarzschild = std::get_if<Schwarzschild>(&solution)) {
+    return schwarzschild->variables(coords, tags);
+  }
+  if (auto* kerrschild = std::get_if<KerrSchild>(&solution)) {
+    return kerrschild->variables(coords, tags);
+  }
+  return tuples::TaggedTuple<RequestedTags...>();
+}
 
+void test_linear_momentum_surface_integral(
+    const double distance, const double mass, const double boost_speed,
+    const std::variant<Schwarzschild, KerrSchild> solution) {
```

Comment:
Same comments as in `Test_AdmMass.cpp`. If it makes things easier you could combine the two files (up to you).